src/Screens/components/ToggleSwitch.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { useState } from 'react';
+import RN from 'react-native';
+
+import * as localization from '../../localization';
+
+import Message from './Message';
+import colors from './colors';
+import fonts from './fonts';
+
+interface Props {
+  initialValue: boolean | undefined;
+  messageOn: localization.MessageId;
+  messageOff: localization.MessageId;
+  messageStyle?: RN.StyleProp<RN.TextStyle>;
+  toggleSwitch: () => boolean;
+  testID?: string;
+}
+
+const ToggleSwitch = ({
+  initialValue,
+  messageOn,
+  messageOff,
+  messageStyle,
+  toggleSwitch,
+  testID,
+}: Props) => {
+  const [isEnabled, setIsEnabled] = useState(initialValue);
+
+  const onPress = () => {
+    if (toggleSwitch()) {
+      setIsEnabled(previousState => !previousState);
+    }
+  };
+
+  return (
+    <RN.View style={styles.container}>
+      <RN.Switch
+        trackColor={{ false: colors.lightGray, true: colors.lighterBlue }}
+        thumbColor={isEnabled ? colors.lightBlue : colors.gray}
+        onValueChange={onPress}
+        value={isEnabled}
+        testID={testID}
+      />
```

Comment:
About this one: Remove the local-state of this component, and use the value of _mentor.isVacationing_ to render whether the switch is in toggled or not.

Like this is fine when the changeVacationStatus-request is successfull, but imagine what happens if the request fails => the state is not in sync with reality anymore.